libraries/stdlib/genglsl/mx_normalmap.glsl

```diff
@@ -14,3 +14,20 @@ void mx_normalmap(vec3 value, int map_space, float normal_scale, vec3 N, vec3 T,
     // Normalize the result.
     result = normalize(value);
 }
+
+void mx_normalmap_vector2(vec3 value, int map_space, vec2 normal_scale, vec3 N, vec3 T,  out vec3 result)
```

Comment:
To my mind, your new generalized `vector2` implementation should now be used by _both_ code paths, with the original `float` version simply calling into this one with the same value in both channels.

This would remove any duplication of code between `vector2` and `float` paths, and make the node more straightforward to maintain over time.

Does that streamlined approach seem reasonable to you as well?

Reply:
Sounds good, I adjusted the implementations accordingly.